probe/probe.go

```diff
@@ -0,0 +1,87 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 Authors of KubeArmor
+
+package probe
+
+
+import (
+	"fmt"
+	"syscall"
+    "strings"
+	"golang.org/x/mod/semver"
+    "io/ioutil"
+    "log"
+
+)
+
+func PrintProbeResult() error{
+    
+    fmt.Printf("Host: \n")
+    fmt.Printf("\t Observability/Audit: %s \n", checkHostAuditSupport())
+    fmt.Printf("\t Enforcement: %s \n", checkHostLsmSupport())
+	return nil
+}
+
+func checkHostAuditSupport() string {
+    hostKernelVersion, err := detectHostKernelVersion()
+	if err != nil {
+		return ""
+	}
+	if hostKernelVersion == false {
+	    return "Not Supported ( Kernel Version < 4.14 )"
+	}else{
+        return "Supported ( Kernel Version >= 4.14 )"
+    }
+}
+
+func checkHostLsmSupport() string {
+    supportedLsm := getHostSupportedLsm()
+    enforcement := "None"
+    if strings.Contains(supportedLsm, "selinux"){
+        enforcement = "Partial"
+    }else if strings.Contains(supportedLsm, "apparmor"){
+        enforcement = "Full"
+    }
+    return enforcement;
+}
+
+func detectHostKernelVersion() (bool, error) {
+	var uname syscall.Utsname
+    if err := syscall.Uname(&uname); err == nil {
+        kernelVersion:= int8ToStr(uname.Release[:])
+        s := strings.Split(kernelVersion, "-")
+        kversion := s[0]
+
+        if semver.Compare(kversion, "4.14") >= 0 {
+            return true, nil
+        }
+	}
+	return false, nil;
+}
+
+func getHostSupportedLsm() string {
+
+    b, err := ioutil.ReadFile("/sys/kernel/security/lsm")
```

Comment:
ioutil was deprecated in 1.16, Let's avoid it.
Ref https://go.dev/doc/go1.16#ioutil